src/v/cluster/service.cc

```diff
@@ -103,4 +107,26 @@ ss::future<configuration_update_reply> service::update_node_configuration(
       });
 }
 
+ss::future<finish_partition_update_reply> service::finish_partition_update(
+  finish_partition_update_request&& req, rpc::streaming_context&) {
+    return ss::with_scheduling_group(
+      get_scheduling_group(), [this, req = std::move(req)]() mutable {
+          return do_finish_partition_update(std::move(req));
+      });
+}
+
+ss::future<finish_partition_update_reply>
+service::do_finish_partition_update(finish_partition_update_request&& req) {
+    auto ec
+      = co_await _topics_frontend.local().finish_moving_partition_replicas(
+        req.ntp,
+        req.new_replica_set,
+        config::shard_local_cfg().replicate_append_timeout_ms()
```

Comment:
we really should move away from this kind of pattern of using the shard local config, it makes it impossible to test things without modifying global state. can you push this to a config object?

Reply:
can I do it as a separate PR ? there are more places like this all over the code.

Reply:
100% sounds good. can you create a ticket for this.